service/debugger/debugger.go

```diff
@@ -916,7 +936,31 @@ func (d *Debugger) Goroutines(start, count int) ([]*api.Goroutine, int, error) {
 	if err != nil {
 		return nil, 0, err
 	}
+
+	// Prioritize running goroutines in the first batch we return (adds up to
+	// 'count' running goroutines to the first batch and insures that they
+	// aren't returned a second time in the batches after the first one).
+	if start > 0 {
```

Comment:
I still don't like the semantics here, and I think it would be better to have a parameter like `onlyRunning` if a caller is only concerned with running Goroutines.

Consider a caller specifies `start == 0, count == 5`, then (and I don't know why they would, but just consider) they call it again with `start == 0, count ==5`. All running goroutines would be removed from the list, which just kind of seems like odd behavior. Lastly, I don't think we should ever return more than `count` items, that's just very unexpected behavior.

Reply:
It doesn't make sense to call `proc.GoroutinesInfo` with arbitrary values of `start`, it will skip an arbitrary number n (with `0 <= n < start`).

You've convinced me that this feature (returning running goroutines first) isn't worth it. I don't think we can implement it without the weird behavior and since running goroutines are returned as part of GetState it isn't that useful anyway.

I've removed this code and added an exetended explanation of Start and Count.